components/Users/FoodAddForm.jsx

```diff
@@ -0,0 +1,225 @@
+  padding: "20px",
+}));
+
+const StyledContainer = styled.div((props) => ({
+  display: "flex",
+  width: "70%",
+  height: "60px",
+  flexDirection: "column",
+  alignItems: "center",
+}));
+
+const StyledField = styled(Field)(() => ({
+  width: "100%",
+  height: "45px",
+  border: "1px solid #ccc",
+  borderRadius: "5px",
+}));
+
+const StyledSelectField = styled(Field)(() => ({
+  width: "100%",
+  height: "45px",
+  border: "1px solid #ccc",
+  borderRadius: "5px",
+}));
+
+const FoodAddForm = (props) => {
+  const [snackbar, setSnackbar] = useState(null);
+  const handleCloseSnackbar = () => setSnackbar(null);
+  let MealDescription = props.MealDescription;
+
+  const saveFoodEntry = async (dataWithUpdatedTime) => {
+    let sessionUser = props.sessionUser;
+    let new_data = {
+      id: "",
+      userEmail: sessionUser.email,
+      ...dataWithUpdatedTime,
+    };
+
+    // api call for adding new food entry with optimistic update
+
+    await props.mutateFoodPerEmail(
+      async (foodEntries) => {
+        let { id, ...dataWithoutId } = new_data;
+        const updatedTodo = await fetch("/api/userFoodEntry", {
+          method: "POST",
+          body: JSON.stringify(dataWithoutId),
+        });
+
+        const addedRow = await updatedTodo.json();
+
+        return [...foodEntries, addedRow];
+      },
+
+      { revalidate: true }
+    );
+
+    setSnackbar({
+      children: "Entry successfully saved",
+      severity: "success",
+    });
+  };
+
+  return (
+    <Paper
+      elevation={10}
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        alignItems: "center",
+        margin: "auto",
+        backgroundColor: "#dae6e5",
+        height: "350px",
+      }}
+    >
+      <Formik
+        initialValues={{
+          foodName: "Food Name",
+          calorieValue: "Calorie Value",
+          takenAt: "",
+          Meal: "Select Meal",
+        }}
+        validationSchema={foodYupSchema}
+        onSubmit={(values, { resetForm }) => {
+          const { takenAt, calorieValue, ...restData } = values;
+          let calValueInt = parseInt(calorieValue);
+          const { mealName, ...exceptMeal } = restData;
+
+          const timeUnix = new Date(String(takenAt));
+          const UpdatedTime = timeUnix.getTime() / 1000;
+          const dataWithUpdatedTime = {
+            ...restData,
+            calorieValue: calValueInt,
+            takenAt: UpdatedTime,
+          };
+
+          saveFoodEntry(dataWithUpdatedTime);
+          resetForm();
+        }}
+      >
+        {({
+          isSubmitting,
+          submitForm,
+          isValid,
+          dirty,
+          errors,
+          touched,
+          values,
+        }) => (
+          <StyledForm>
+            <StyledContainer err={true}>
+              <StyledField name="foodName" />
+
+              {!errors.foodName && touched.foodName ? (
+                <CheckCircleIcon style={{ color: "#46eb67" }} />
+              ) : null}
```

Comment:
Try to avoid using the ternary operator.
```
{!errors.foodName && touched.foodName &$ (
                <CheckCircleIcon style={{ color: "#46eb67" }} />
              ) 
}

```